source/features/reactions-avatars.tsx

```diff
@@ -56,7 +54,26 @@ function getParticipants(button: HTMLButtonElement): Participant[] {
 	return participants;
 }
 
-async function showAvatarsOn(commentReactions: Element): Promise<void> {
+const viewportObserver = new IntersectionObserver(changes => {
+	for (const change of changes) {
+		if (change.isIntersecting) {
+			showAvatarsOn(change.target);
+			viewportObserver.unobserve(change.target);
+		}
+	}
+}, {
+	// Start loading a little before they become visible
+	rootMargin: '500px',
+});
+
+const resizeObserver = new ResizeObserver(([{target}], observer) => {
+	if (!target.isConnected) {
+		observer.unobserve(target);
+		observeReactions();
+	}
+});
```

Comment:
Replaced the usage of `onElementRemoval` on every reaction element, with a global `ResizeObserver`.

This makes the feature much easier to deinit, and should make it lighter too.